packages/open-truss/src/utils/descibe-zod.ts

```diff
@@ -0,0 +1,55 @@
+import { type z } from 'zod'
+import { BaseOpenTrussComponentV1PropsShape } from '../configuration'
+import { type YamlType } from '../utils/yaml'
+
+type ZodShape = ZodDescriptionObject | string[]
+
+export interface ZodDescriptionObject {
+  type?: string
+  defaultValue?: YamlType
+  shape?: ZodShape
+}
+
+// describeZod outputs a readable summary of a zod type.
+// In Open Truss we use this to describe a component's prop types
+// so that we can create a UX to guide filling those props in.
+export function describeZod(
+  zobject: object,
+): Record<string, ZodDescriptionObject> {
+  return Object.entries(zobject).reduce<Record<string, ZodDescriptionObject>>(
+    (description, [key, value]) => {
+      // We don't process the base props like config and data.
+      if (Object.keys(BaseOpenTrussComponentV1PropsShape.shape).includes(key)) {
```

Comment:
It does feel like `describe-zod` could be an isolated function that you pass in dependencies to like `BaseOpenTrussComponentV1PropsShape`. But I don't think that would get us much right now, so this looks 👍 to me.